smac/runhistory/runhistory.py

```diff
@@ -282,24 +296,33 @@ def get_cost(self, config: Configuration):
         config_id = self.config_ids[config]
         return self.cost_per_config.get(config_id, np.nan)
 
-    def get_runs_for_config(self, config: Configuration):
+    def get_runs_for_config(self, config: Configuration, max_budget: bool = True):
         """Return all runs (instance seed pairs) for a configuration.
 
         Parameters
         ----------
         config : Configuration from ConfigSpace
             Parameter configuration
-
+        max_budget : bool
+            Select only the max budget run from each configuration (default=True)
         Returns
         -------
-        instance_seed_pairs : list<tuples of instance, seed>
+        instance_seed_budget_pairs : list<tuples of instance, seed, budget>
         """
         config_id = self.config_ids.get(config)
-        return self._configid_to_inst_seed.get(config_id, [])
+        runs = self._configid_to_inst_seed_budget.get(config_id, {}).copy()
+
+        # select only the max budget run if specified
+        if max_budget:
+            for k, v in runs.items():
+                runs[k] = [max(v)]
+
+        # convert to inst-seed-budget key
+        runs = [InstSeedBudgetKey(k.instance, k.seed, budget) for k, v in runs.items() for budget in v]
+        return runs
 
     def get_instance_costs_for_config(self, config: Configuration):
```

Comment:
Shouldn't this also get a flag to only return the cost on the maximum budget?

Reply:
I'm not sure if that will be right, because we cannot combine the costs of different (runtime) budgets into 1. So this by default takes only the max budget. Should I give an option of returning instance costs for each instance-budget combination?

Reply:
Ah, because the `self.get_runs_for_config` only returns the cost on the highest budget, right?

Reply:
Yes, that is the default behaviour.